lib/mayaUsd/render/mayaToHydra/renderOverride.cpp

```diff
@@ -770,6 +848,110 @@ bool MtohRenderOverride::nextRenderOperation() {
     return ++_currentOperation < static_cast<int>(_operations.size());
 }
 
+#if MAYA_API_VERSION >= 20210000
+bool MtohRenderOverride::select(
+    const MHWRender::MFrameContext& frameContext,
+    const MHWRender::MSelectionInfo& selectInfo,
+    bool /*useDepth*/,
+    MSelectionList& selectionList,
+    MPointArray& worldSpaceHitPts)
+{
+    MStatus status = MStatus::kFailure;
+
+    MMatrix viewMatrix = frameContext.getMatrix(MHWRender::MFrameContext::kViewMtx, &status);
+    if (status != MStatus::kSuccess)
+        return false;
+
+    MMatrix projMatrix = frameContext.getMatrix(MHWRender::MFrameContext::kProjectionMtx, &status);
+    if (status != MStatus::kSuccess)
+        return false;
+
+    int view_x, view_y, view_w, view_h;
+    status = frameContext.getViewportDimensions(view_x, view_y, view_w, view_h);
+    if (status != MStatus::kSuccess)
+        return false;
+
+    unsigned int sel_x, sel_y, sel_w, sel_h;
+    status = selectInfo.selectRect(sel_x, sel_y, sel_w, sel_h);
+    if (status != MStatus::kSuccess)
+        return false;
+
+    // Compute a pick matrix that, when it is post-multiplied with the projection matrix, will
+    // cause the picking region to fill the entire/ viewport for OpenGL selection.
+    {
+        MMatrix pickMatrix;
+        pickMatrix[0][0] = view_w / double(sel_w);
+        pickMatrix[1][1] = view_h / double(sel_h);
+        pickMatrix[3][0] = (view_w - (double)(sel_x * 2 + sel_w)) / double(sel_w);
+        pickMatrix[3][1] = (view_h - (double)(sel_y * 2 + sel_h)) / double(sel_h);
+
+        projMatrix *= pickMatrix;
+    }
+
+    const bool pointSnappingActive = selectInfo.pointSnapping();
+
+    // Set up picking params.
+    HdxPickTaskContextParams pickParams;
+    pickParams.resolution.Set(view_w, view_h);
+    pickParams.viewMatrix.Set(viewMatrix.matrix);
+    pickParams.projectionMatrix.Set(projMatrix.matrix);
+    pickParams.resolveMode = HdxPickTokens->resolveUnique;
```

Comment:
Maybe, but I haven't seen issue for single selection when testing the code. With the code merged, I am sure there are still some improvements in the plugin side as we iterate over time. So maybe something for future?

Reply:
Well, like I said, this falls more under the class of an optimization.  It's a pretty easy check to make, and you'd then be able to skip the `ResolveUniqueHits_Workaround` if you were doing `resolveNearestToCenter` or `resolveNearestToCamera`... but you're likely also right, in that it may not be noticeable for single select, since the pick region is generally going to be pretty small. So your call.